server/pkg/endpoints/node.go

```diff
@@ -48,17 +44,29 @@ func ForNode(tx *proxystore.Tx, si *localnetv1.ServiceInfo, nodeName string) (in
 	tx.EachEndpointOfService(svc.Namespace, svc.Name, func(info *localnetv1.EndpointInfo) {
 		info = proto.Clone(info).(*localnetv1.EndpointInfo)
 
-		epNodeName := info.NodeName
-		if epNodeName == "" {
-			epNodeName = info.Topology[hostnameLabel]
-		}
-
-		info.Endpoint.Local = epNodeName == nodeName
+		info.Endpoint.Local = info.Topology.Node == nodeName
 
 		if !info.Conditions.Ready {
 			return
 		}
 
+		if hints := info.Hints; hints != nil {
+			if len(hints.Zones) != 0 {
```

Comment:
@robscott hi, can you confirm it's the spirit of the zones hint? :)

Reply:
I'm not that familiar with the code base here, but this looks reasonable in isolation. With that said, there's more logic here when it comes to handling the interactions with traffic policies - refer to the recently refactored upstream logic here: https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/topology.go#L40

Reply:
thanks @robscott I'll open an issue to dig this